File: top_interview_questions/1_easy/8_math/fizz_buzz/solution.py
class Solution:
    def fizzBuzz(self, n: int) -> List[str]:
        result = []

        for i in range(1, n + 1):
            if i % 15 == 0:
                string = "FizzBuzz"
            elif i % 5 == 0:
                string = "Buzz"
            elif i % 3 == 0:
                string = "Fizz"
            else:
                string = str(i)

            result.append(string)

        return result

File: top_interview_questions/1_easy/8_math/fizz_buzz/solution.go
package main

import (
	"fmt"
	"strconv"
)

func main() {
	fmt.Println(fizzBuzz(20))
}

func fizzBuzz(n int) []string {
	result := []string{}

	for i := 1; i <= n; i++ {
		if i%15 == 0 {
			result = append(result, "FizBuzz")
		} else if i%5 == 0 {
			result = append(result, "Buzz")
		} else if i%3 == 0 {
			result = append(result, "Fiz")
		} else {
			result = append(result, strconv.Itoa(i))
		}
	}

	return result
}
